FurCoNZ/Services/Payment/StripePaymentProvider.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Stripe.Checkout;
+using Stripe;
+
+using FurCoNZ.DAL;
+using FurCoNZ.Models;
+using Microsoft.Extensions.Options;
+using FurCoNZ.Configuration;
+
+namespace FurCoNZ.Services.Payment
+{
+    public class StripeService : IPaymentProvider
+    {
+        private readonly FurCoNZDbContext _dbContext;
+        private readonly IOptions<StripeSettings> _options;
+        private readonly IOrderService _orderService;
+        private readonly ILogger _logger;
+        private readonly PaymentIntentService _paymentIntentService;
+
+        public string Name => "Stripe";
+
+        public string SupportedMethods => "Credit Card";
+
+        public string Description => "Online payment processor for credit cards.";
+
+        public StripeService(FurCoNZDbContext dbContext, IOptions<StripeSettings> options, IOrderService orderService, ILogger<StripeService> logger)
+        {
+            _dbContext = dbContext;
+            _options = options;
+            _orderService = orderService;
+            _logger = logger;
+            _paymentIntentService = new PaymentIntentService();
+        }
+
+        public async Task AddStripeSessionToOrderAsync(int orderId, string sessionId, CancellationToken cancellationToken = default)
+        {
+            var stripeSession = new StripeSession
+            {
+                Id = sessionId,
+                State = StripeSessionState.None,
+                OrderId = orderId,
+            };
+
+            await _dbContext.StripeSessions.AddAsync(stripeSession, cancellationToken);
+            await _dbContext.SaveChangesAsync(cancellationToken);
+        }
+
+        public async Task FufillOrderFromCompletedSessionAsync(Session session, CancellationToken cancellationToken = default)
+        {
+            var stripeSession = await _dbContext.StripeSessions
+                .SingleOrDefaultAsync(s => s.Id == session.Id, cancellationToken);
```

Comment:
`.SingleOrDefaultAsync()` has a performance hit over `.FirstOrDefaultAsync()`, since it has to scan the entire table (it has to `throw` if more than one record is found). Since this is a primary key, it shouldn't be possible to have more than one record anyway.
```suggestion
                .FirstOrDefaultAsync(s => s.Id == session.Id, cancellationToken);
```